### PR TITLE
feat(argo-workflow): Upgrade to 3.0.0

### PR DESCRIPTION
**Pull request by goock**
## Why change 

To upgrade Argo Workflow to the latest version `3.0.0`

## Description

This PR is a draft until the final `3.0.0` is released.
This PR allows testing recent release candidates.

## Release history
- `3.0.2`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.2
- `3.0.1`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.1
- `3.0.0`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0
- `3.0.0-rc9`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc9
- `3.0.0-rc8`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc8
- `3.0.0-rc6`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc6
- `3.0.0-rc5`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc5
- `3.0.0-rc4`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc4
- `3.0.0-rc3`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc3
- `3.0.0-rc2`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc2
- `3.0.0-rc1`: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc1

## New features in manifest
-  env var `LEADER_ELECTION_IDENTITY`: https://github.com/argoproj/argo-cd/issues/3073
- `lease` resource

## Ref

- https://github.com/coreweave/infra-pm/issues/14
- https://github.com/argoproj/argo/pull/4679

# TODO
- [x] - `FAIL HOSTS` - does not display, https://github.com/argoproj/argo/issues/5054
- [x] - test concierge in own namespace
- [x] - wait for final release: https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0
- [x] - retest final version with concierge
- [x] - apply to concierge https://github.com/coreweave/concierge-backend/pull/477
- [ ] - apply https://github.com/coreweave/kube-charts/pull/30

### CODE REVIEW

**Comment by goock** [comment]
There is an issue with node names. It must be related to our manifest, it works in local dev with full manifest + no one else does not complain about this issue.
The issue comes from this line:
https://github.com/argoproj/argo-workflows/blob/8a8759f38c5773cebecf3dd5fca1ec3b5095ef27/workflow/controller/operator.go#L1084
underneath there is a configmaps's cache. Where the node name is saved while the workflow is processed.

## Ref
https://github.com/argoproj/argo-workflows/issues/5054

File: argo-workflow/argo-install.yaml [hunk 1]
```diff
@@ -177,15 +185,11 @@ kind: ConfigMap
 metadata:
   name: workflow-controller-configmap
 data:
-  config: |
```
Comment (goock) [inline argo-workflow/argo-install.yaml:180]: This is a new format of the ConfigMaps https://github.com/argoproj/argo-workflows/blob/master/docs/workflow-controller-configmap.yaml

File: argo-workflow/argo-install.yaml [hunk 2]
```diff
@@ -177,15 +185,11 @@ kind: ConfigMap
 metadata:
   name: workflow-controller-configmap
 data:
-  config: | 
-    containerRuntimeExecutor: k8sapi
-    featureFlags:
-      resourcesDuration: true
```
Comment (goock) [inline argo-workflow/argo-install.yaml:183]: it's gone from configmaps in v3

**Comment by goock** [comment]
@salanki @brianventuro Despite it's not the final version I think we can use it in Concierge.
- Eventually, I found a solution for missing hostnames - they missing when the pod finishes too quickly.
- I tested with concierge in local dev, we only need to change `containerRuntimeExecutor: docker`

File: argo-workflow/argo-install.yaml [hunk 3]
```diff
@@ -289,10 +293,16 @@ spec:
         - --configmap
         - workflow-controller-configmap
         - --executor-image
-        - argoproj/argoexec:v2.11.7
+        - argoproj/argoexec:v3.0.0-rc3
         - --namespaced
         command:
         - workflow-controller
+        env:
+        - name: LEADER_ELECTION_IDENTITY
```
Comment (mecampbellsoup) [inline argo-workflow/argo-install.yaml:301]: What's this?

Comment (goock) [reply argo-workflow/argo-install.yaml:301]: you can have more controllers and this env identifies the leader.

**Comment by salanki** [comment]
Once 3.0 is released, let's turn this into a Helm chart and add to kube-charts as well.

**Comment by goock** [comment]
> Once 3.0 is released, let's turn this into a Helm chart and add to kube-charts as well.

yup, I will https://github.com/coreweave/kube-charts/issues/5#event-4409486301

**Comment by goock** [comment]
Version 3.0.0 has an issue creating argo server: `Readiness probe failed: Get http://10.140.155.237:2746/: EOF`

File: argo-workflow/argo-install.yaml [hunk 4]
```diff
@@ -240,9 +244,8 @@ spec:
       - args:
         - server
         - --namespaced
-#        - --auth-mode 
-#        - client
-        image: argoproj/argocli:v2.11.7
+        - --secure=false
```
Comment (goock) [inline argo-workflow/argo-install.yaml:247]: TLS is enabled by default since 3.0.0
https://github.com/argoproj/argo-workflows/releases/tag/v3.0.1

**Review by mecampbellsoup — COMMENTED** [review]
So, is this `argo-install.yaml` shown on our docs as well as used by us internally? I don't really get what this repo contains `kubernetes-cloud`, I've never worked in here. I don't think I am in a good position to review this PR.

**Review by mecampbellsoup — APPROVED** [review]

**Comment by goock** [comment]
@cesarmesones This is an upgrade of Argo workflow. I implemented a special feature which allows retrying step on a different host. This feature helps us not repeat the same render job on the faulty host. This is the case, especially for Blender's Eevee engine.

Despite approval from your side, we would need to schedule an upgrade in concierge (tenant-concierge).
I already tested the new Argo in my namespace with the concierge.
The original manifest consists of 2 major kinds, those namespaced and those cluster wide. The namespaced part is the only in the manifest here, the cluster-wide part was applied by @salanki and he does not remember exactly what was applied.
I would need your assist to upgrade Argo in `tenant-concierge` in case something will go wrong.

If you have any questions let me know here or via Slack.

**Comment by salanki** [comment]
This looks good to me. How do you envision the upgrade of concierge to look like, just apply this manifest? We want to turn this into a Helm chart as well and throw it in `kube-charts`.

**Comment by goock** [comment]
> This looks good to me. How do you envision the upgrade of concierge to look like, just apply this manifest? We want to turn this into a Helm chart as well and throw it in `kube-charts`.

I already did it in https://github.com/coreweave/kube-charts/pull/30. I just needed to update to latest ver. 3.0.2. Waiting for review.

**Comment by cesarmesones** [comment]
this looks good to me too for just a straight manual yaml update unless @salanki wants you to use the helm chart youve created to do the upgrade

**Comment by goock** [comment]
> this looks good to me too for just a straight manual yaml update unless @salanki wants you to use the helm chart youve created to do the upgrade

I think we could upgrade concierge with manual yaml, which does not have `ingress` and this is exactly what we have right now in concierge. If something goes wrong we always can callback with current version.

**Pull request merged**